i18n script doesn't work (on Windows?)

### Steps to reproduce

1. Clone the project
2. Add a string somewhere
3. `yarn i18n`
4. Get errors (plural)

### Outcome

#### What did you expect?

For the translations file to be updated.

#### What happened instead?

First, I needed to `yarn add -D @babel/highlight` so the script would run at all:

```
PS D:\Workspaces\element-web-stack\element-web> yarn i18n
yarn run v1.22.22
$ matrix-gen-i18n src res packages/shared-components/src && yarn i18n:sort && yarn i18n:lint
node:internal/modules/cjs/loader:1423
  throw err;
  ^

Error: Cannot find module '@babel/highlight'
Require stack:
- D:\Workspaces\element-web-stack\element-web\node_modules\matrix-web-i18n\node_modules\@babel\code-frame\lib\index.js
- D:\Workspaces\element-web-stack\element-web\node_modules\matrix-web-i18n\node_modules\@babel\traverse\lib\path\replacement.js
- D:\Workspaces\element-web-stack\element-web\node_modules\matrix-web-i18n\node_modules\@babel\traverse\lib\path\index.js
- D:\Workspaces\element-web-stack\element-web\node_modules\matrix-web-i18n\node_modules\@babel\traverse\lib\context.js
- D:\Workspaces\element-web-stack\element-web\node_modules\matrix-web-i18n\node_modules\@babel\traverse\lib\traverse-node.js
- D:\Workspaces\element-web-stack\element-web\node_modules\matrix-web-i18n\node_modules\@babel\traverse\lib\path\context.js
- D:\Workspaces\element-web-stack\element-web\node_modules\matrix-web-i18n\node_modules\@babel\traverse\lib\index.js
- D:\Workspaces\element-web-stack\element-web\node_modules\matrix-web-i18n\scripts\gen-i18n.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1420:15)
    at defaultResolveImpl (node:internal/modules/cjs/loader:1058:19)
    at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1063:22)
    at Module._load (node:internal/modules/cjs/loader:1226:37)
    at TracingChannel.traceSync (node:diagnostics_channel:328:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:244:24)
    at Module.require (node:internal/modules/cjs/loader:1503:12)
    at require (node:internal/modules/helpers:152:16)
    at Object.<anonymous> (D:\Workspaces\element-web-stack\element-web\node_modules\matrix-web-i18n\node_modules\@babel\code-frame\lib\index.js:8:18)
    at Module._compile (node:internal/modules/cjs/loader:1760:14) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'D:\\Workspaces\\element-web-stack\\element-web\\node_modules\\matrix-web-i18n\\node_modules\\@babel\\code-frame\\lib\\index.js',
    'D:\\Workspaces\\element-web-stack\\element-web\\node_modules\\matrix-web-i18n\\node_modules\\@babel\\traverse\\lib\\path\\replacement.js',
    'D:\\Workspaces\\element-web-stack\\element-web\\node_modules\\matrix-web-i18n\\node_modules\\@babel\\traverse\\lib\\path\\index.js',
    'D:\\Workspaces\\element-web-stack\\element-web\\node_modules\\matrix-web-i18n\\node_modules\\@babel\\traverse\\lib\\context.js',
    'D:\\Workspaces\\element-web-stack\\element-web\\node_modules\\matrix-web-i18n\\node_modules\\@babel\\traverse\\lib\\traverse-node.js',
    'D:\\Workspaces\\element-web-stack\\element-web\\node_modules\\matrix-web-i18n\\node_modules\\@babel\\traverse\\lib\\path\\context.js',
    'D:\\Workspaces\\element-web-stack\\element-web\\node_modules\\matrix-web-i18n\\node_modules\\@babel\\traverse\\lib\\index.js',
    'D:\\Workspaces\\element-web-stack\\element-web\\node_modules\\matrix-web-i18n\\scripts\\gen-i18n.js'
  ]
}

Node.js v24.10.0
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Then, I needed to install `jq` (which appears to be a hidden dependency?):

```
Wrote 3531 strings to src/i18n/strings/en_EN.json
$ jq --sort-keys '.' src/i18n/strings/en_EN.json > src/i18n/strings/en_EN.json.tmp && mv src/i18n/strings/en_EN.json.tmp src/i18n/strings/en_EN.json
'jq' is not recognized as an internal or external command,
operable program or batch file.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

*Then*, I needed to fix the `package.json` entry for the `i18n:sort` script because `'.'` was not recognized properly (possible due to me being on Windows). The fix was to swap it for `\".\"` (escaping required because json). The error was:

```
Wrote 3531 strings to src/i18n/strings/en_EN.json
$ jq --sort-keys '.' src/i18n/strings/en_EN.json > src/i18n/strings/en_EN.json.tmp && mv src/i18n/strings/en_EN.json.tmp src/i18n/strings/en_EN.json
jq: error: syntax error, unexpected INVALID_CHARACTER, expecting end of file at <top-level>, line 1, column 1:
    '.'
    ^
jq: 1 compile error
error Command failed with exit code 3.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 3.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Then finally, I could see my string added.

### Operating system

Windows 11 Pro

### Browser information

Chrome something

### URL for webapp

localhost:8080

### Application version

develop+commits

### Homeserver

localhost:8008

### Will you send logs?

No

--- a/package.json
+++ b/package.json
@@ -27,7 +27,8 @@
         "matrix-gen-i18n": "scripts/gen-i18n.cjs",
         "matrix-compare-i18n-files": "scripts/compare-file.cjs",
         "matrix-i18n-rekey": "scripts/rekey.cjs",
-        "matrix-i18n-lint": "scripts/lint-i18n.cjs"
+        "matrix-i18n-lint": "scripts/lint-i18n.cjs",
+        "matrix-sort-i18n": "scripts/sort-json.cjs"
     },
     "scripts": {
         "build:ts": "tsc",

--- a/scripts/sort-json.cts
+++ b/scripts/sort-json.cts
@@ -1,5 +1,5 @@
 /*
-Copyright 2026 New Vector Ltd
+Copyright 2026 Aditya Cherukuru
 
 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.

--- a/scripts/sort-json.cts
+++ b/scripts/sort-json.cts
@@ -1,0 +1,97 @@
+/*
+Copyright 2026 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/**
+ * Cross-platform JSON key sorting utility.
+ * Replaces the need for `jq --sort-keys` which has issues on Windows due to:
+ * - jq not being installed by default
+ * - Shell quoting differences ('.' vs ".")
+ *
+ * Usage: matrix-sort-i18n <file.json> [file2.json ...]
+ */
+
+import * as fs from "fs";
+import * as path from "path";
+
+/**
+ * Recursively sorts the keys of an object alphabetically.
+ * Arrays are preserved as-is (not sorted), but objects within arrays are sorted.
+ */
+function sortObjectKeys(obj: unknown): unknown {
+    if (obj === null || typeof obj !== "object") {
+        return obj;
+    }
+
+    if (Array.isArray(obj)) {
+        return obj.map(sortObjectKeys);
+    }
+
+    const sortedObj: Record<string, unknown> = {};
+    const keys = Object.keys(obj as Record<string, unknown>).sort();
+    for (const key of keys) {
+        sortedObj[key] = sortObjectKeys((obj as Record<string, unknown>)[key]);
+    }
+    return sortedObj;
+}
+
+/**
+ * Sorts JSON file keys and writes back to the same file.
+ */
+function sortJsonFile(filePath: string): void {
+    const absolutePath = path.resolve(filePath);
+
+    if (!fs.existsSync(absolutePath)) {
+        throw new Error(`File not found: ${absolutePath}`);
+    }
+
+    const content = fs.readFileSync(absolutePath, "utf8");
+    let parsed: unknown;
+
+    try {
+        parsed = JSON.parse(content);
+    } catch (e) {
+        throw new Error(`Invalid JSON in file: ${absolutePath}`);
+    }
+
+    const sorted = sortObjectKeys(parsed);
+    const output = JSON.stringify(sorted, null, 4) + "\n";
+
+    fs.writeFileSync(absolutePath, output, "utf8");
+    console.log(`Sorted keys in: ${filePath}`);
+}
+
+// Main execution
+if (process.argv.length < 3) {
+    console.error("Usage: matrix-sort-i18n <file.json> [file2.json ...]");
+    console.error("Sorts JSON object keys alphabetically in-place.");
+    process.exit(1);
+}
+
+const files = process.argv.slice(2);
+let hasError = false;
+
+for (const file of files) {
+    try {
+        sortJsonFile(file);
+    } catch (e) {
+        console.error(`Error processing ${file}:`, (e as Error).message);
+        hasError = true;
+    }
+}
+
+if (hasError) {
+    process.exit(1);
+}

